manager/integration/tests/test_orphan.py

```diff
@@ -0,0 +1,940 @@
+                                         mkfs_ext4_options)
+    return disk_path
+
+
+def cleanup_host_disk(client, *args):  # NOQA
+    # clean disk
+    for vol_name in args:
+        # umount disk
+        common.cleanup_host_disk(vol_name)
+        # clean volume
+        cleanup_volume(client, vol_name)
+
+
+def create_volume(client, vol_name, size, node_id, r_num):  # NOQA
+    volume = client.create_volume(name=vol_name, size=size,
+                                  numberOfReplicas=r_num)
+    assert volume.numberOfReplicas == r_num
+    assert volume.frontend == VOLUME_FRONTEND_BLOCKDEV
+
+    volume = common.wait_for_volume_detached(client, vol_name)
+    assert len(volume.replicas) == r_num
+
+    assert volume.state == "detached"
+    assert volume.created != ""
+
+    volumeByName = client.by_id_volume(vol_name)
+    assert volumeByName.name == volume.name
+    assert volumeByName.size == volume.size
+    assert volumeByName.numberOfReplicas == volume.numberOfReplicas
+    assert volumeByName.state == volume.state
+    assert volumeByName.created == volume.created
+
+    volume.attach(hostId=node_id)
+    volume = wait_for_volume_healthy(client, vol_name)
+
+    return volume
+
+
+def cleanup_volume(client, vol_name):  # NOQA
+    volume = client.by_id_volume(vol_name)
+    volume.detach(hostId="")
+    client.delete(volume)
+    common.wait_for_volume_delete(client, vol_name)
+
+
+@pytest.mark.orphan
+def test_orphaned_dirs_with_wrong_naming_format(client, volume_name):  # NOQA
+    """
+    Test orphan CRs are not created for the orphaned directories with wrong
+    naming formats
+    1. Create a new disk holding valid and invalid orphaned replica
+       directories
+    2. Create a volume and then attach to the current node
+    3. Create one valid orphaned replica directories by copying the active
+       replica directory
+    4. Create multiple invalid orphan replica directories with wrong naming
+       format
+    5. Verify orphan list only contains the orphan CR for valid orphaned
+       replica directory
+    """
+
+    lht_hostId = get_self_host_id()
+
+    cleanup_node_disks(client, lht_hostId)
+
+    new_disk_volume_name = 'new-disk-' + generate_random_id(8)
+    new_disk_path = create_host_disk(client,
+                                     new_disk_volume_name,
+                                     str(Gi),
+                                     lht_hostId)
+    new_disk = {"path": new_disk_path, "allowScheduling": True}
+
+    node = client.by_id_node(lht_hostId)
+
+    update_disks = get_update_disks(node.disks)
+    update_disks[new_disk_volume_name] = new_disk
+    node = node.diskUpdate(disks=update_disks)
+    node = wait_for_disk_update(client, node.name, len(update_disks))
+
+    nodes = client.list_node()
+    volume = create_and_check_volume(client, volume_name, len(nodes), SIZE)
+    volume.attach(hostId=lht_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
```

Comment:
This seems repeated in other test function as well, so I would suggest to add a helper function in this test module.

Reply:
Addressed.